SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp

```diff
@@ -162,12 +158,30 @@ void USpatialReceiver::OnAddComponent(const Worker_AddComponentOp& Op)
 	case SpatialConstants::CLIENT_RPC_ENDPOINT_COMPONENT_ID_LEGACY:
 	case SpatialConstants::SERVER_RPC_ENDPOINT_COMPONENT_ID_LEGACY:
 	case SpatialConstants::SERVER_TO_SERVER_COMMAND_ENDPOINT_COMPONENT_ID:
-	case SpatialConstants::AUTHORITY_INTENT_COMPONENT_ID:
 	case SpatialConstants::CLIENT_ENDPOINT_COMPONENT_ID:
 	case SpatialConstants::SERVER_ENDPOINT_COMPONENT_ID:
 	case SpatialConstants::SPATIAL_DEBUGGING_COMPONENT_ID:
 		// Ignore static spatial components as they are managed by the SpatialStaticComponentView.
 		return;
+	case SpatialConstants::UNREAL_METADATA_COMPONENT_ID:
+		// The unreal metadata component is used to indicate when an actor needs to be created from the entity.
+		// This means we need to be inside a critical section, otherwise we may not have all the requisite information at the point of creating the actor.
+		check(bInCriticalSection);
+		PendingAddActors.Emplace(Op.entity_id);
+		return;
+	case SpatialConstants::AUTHORITY_INTENT_COMPONENT_ID:
+		// Check to see if we were waiting for this component for load balancing.
+		if (PendingAclAuthorityChanges.Contains(Op.entity_id))
+		{
+			// Check if we still have authority over the ACL. We should, unless the previously seen authority change was just a flicker.
+			if (StaticComponentView->HasAuthority(Op.entity_id, SpatialConstants::ENTITY_ACL_COMPONENT_ID) || LoadBalanceEnforcer != nullptr)
```

Comment:
Should the || be an && ?

Reply:
yep, thanks, good catch

Reply:
this seems to be the only logical change, which is definitely correct, but now the handover gym has just stopped working again, so back to debugging